src/Test/L0/SecretMaskerTests/LoggedSecretMaskerL0.cs

```diff
@@ -1,28 +1,49 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
-using Agent.Sdk.Util;
-using System;
+
 using Agent.Sdk.SecretMasking;
+
+using SecretMaskerVSO = Microsoft.TeamFoundation.DistributedTask.Logging.SecretMasker;
+using ISecretMaskerVSO = Microsoft.TeamFoundation.DistributedTask.Logging.ISecretMasker;
+
 using Xunit;
 
 namespace Microsoft.VisualStudio.Services.Agent.Tests
 {
-    public class LoggedSecretMaskerL0 : IDisposable
+    public class BuiltInLoggedSecretMaskerL0 : LoggedSecretMaskerTestsBase
+    {
+        protected override ISecretMaskerVSO InitializeSecretMasker()
+        {
+            return new BuiltInSecretMasker();
+        }
+    }
+
+    public class OssLoggedSecretMaskerL0 : LoggedSecretMaskerTestsBase
     {
-        SecretMasker _secretMasker;
-        private bool disposedValue;
+        protected override ISecretMaskerVSO InitializeSecretMasker()
+        {
+            return new OssSecretMasker();
+        }
+    }
 
-        public LoggedSecretMaskerL0()
+    public class LoggedSecretMaskerVSOL0 : LoggedSecretMaskerTestsBase
+    {
+        protected override ISecretMaskerVSO InitializeSecretMasker()
         {
-            _secretMasker = new SecretMasker();
+            return new SecretMaskerVSO();
         }
+    }
+
+    public abstract class LoggedSecretMaskerTestsBase
+    {
+        protected abstract ISecretMaskerVSO InitializeSecretMasker();
```

Comment:
Consider rename to `CreateSecretMasker`. To me, Initialize suggests an operation on an existing instance rather than construction of a new instance.

Reply:
Do it.